torch_geometric/loader/neighbor_loader.py

```diff
@@ -147,7 +152,8 @@ def _set_num_neighbors_and_num_hops(self, num_neighbors):
         # Add at least one element to the list to ensure `max` is well-defined
         self.num_hops = max([0] + [len(v) for v in num_neighbors.values()])
 
-    def _sparse_neighbor_sample(self, index: Tensor):
+    def _sparse_neighbor_sample(self, index: Union[List[int], Tensor]):
```

Comment:
Why? Don't we convert to tensors beforehand?

Reply:
Ah because that made the logic in `__call__` quite confusing. You would have first to check whether or not it was hetero and then if it was handling the conversion separately if it was a mixed input. To me it seemed simpler to handle these cases in the individual functions, as it was actually less code.